apps/aecore/test/aecore_sync_SUITE.erl

```diff
@@ -0,0 +1,460 @@
+    case lists:keytake(K, 1, L) of
+        false ->
+            {Def, L};
+        {value, {_, V}, Rest} ->
+            {V, Rest}
+    end.
+
+%%
+%% Patches:
+%% - use fast mining NIF (aec_pow_cuckoo.erl)
+%% - turn off continuous mining in aec_miner.erl
+%%
+patch_files(Config) ->
+    ok = parse_trans_mod:transform_module(
+           aec_miner,
+           fun miner_xform/2,
+           [{pt_pp_src, true}]),
+    ok = parse_trans_mod:transform_module(
+           aec_pow_cuckoo,
+           fun pow_cuckoo_xform/2,
+           [{pt_pp_src, true}]),
+    move_files_and_compile(["aec_miner.xfm",
+                            "aec_pow_cuckoo.xfm"], Config).
+
+move_files_and_compile(Files, Config) ->
+    PrivDir = ?config(priv_dir, Config),
+    ok = filelib:ensure_dir(filename:join(PrivDir, "foo")),
+    FromDir = filename:join(?config(top_dir, Config),
+                            "_build/test/lib/aecore/src"),
+    lists:foreach(
+      fun(F) ->
+              Base = filename:basename(F, ".xfm"),
+              ErlF = Base ++ ".erl",
+              Src = filename:join(FromDir, F),
+              Tgt = filename:join(PrivDir, ErlF),
+              ct:log("moving ~s~n -> ~s~n", [Src, Tgt]),
+              ok = file:rename(Src, Tgt),
+              cmd(["(cd ", PrivDir, " && erlc -W ", ErlF, ")"])
+      end, Files).
+
+
+miner_xform(Forms, _Opts) ->
+    parse_trans:plain_transform(fun miner_xform/1, Forms).
+
+miner_xform({function,L,init,1,_}) ->
+    L1 = L+1, L2 = L+2,
+    {function,L,init,1,
+     [{clause,L1,
+       [{var,L1,'_'}],
+       [],
+       [{tuple,L2,
+         [{atom,L2,ok},
+          {atom,L2,idle},
+          {record,L2,state,[]}]}]}]};
+miner_xform({function,L,running,3,Clauses}) ->
+    Clauses1 =
+        lists:map(
+          fun({clause,_,[{atom,_,cast},{atom,_,mine},_],_,_} = C) ->
+                  [C1] =
+                      parse_trans:plain_transform(fun miner_xform_1/1, [C]),
+                  C1;
+             (Clause) ->
+                  Clause
+          end, Clauses),
+    {function,L,running,3,Clauses1};
+miner_xform(_) ->
+    continue.
+
+%% Remove all calls to gen_statem:cast(aec_miner, mine) from the
+%% 'running' state - i.e. don't automatically re-mine
+miner_xform_1({call,L,
+               {remote,_,{atom,_,gen_statem},{atom,_,cast}},
+               [{atom,_,aec_miner},{atom,_,mine}]}) ->
+    {atom,L,ok};
+miner_xform_1({clause,_,[{tuple,_,[{atom,_,error},
+                                   {atom,_,generation_count_exhausted}]}],
+               _,_} = C) ->
+    %% don't change next state to idle!
+    C;
+miner_xform_1({tuple,L,[{atom,_,next_state},{atom,_,running},{var,_,S}]}) ->
+    {tuple,L,[{atom,L,next_state},{atom,L,idle},{var,L,S}]};
+miner_xform_1(_) ->
+    continue.
+
+pow_cuckoo_xform(Forms, _Options) ->
+    parse_trans:plain_transform(fun pow_cuckoo_xform/1, Forms).
+
+pow_cuckoo_xform({string,L,"aec_pow_cuckoo28_nif"}) ->
+    {string,L,"aec_pow_cuckoo20_nif"};
```

Comment:
@uwiger  @akorosmezey  @sennui *If* package is meant to include both nif versions, shall this be controlled by configuration rather than parse transform?

Reply:
@lucafavatella currently, both versions _do_ seem to be included, but I agree.
In many cases, it would also be very convenient to be able to select the faster algorithm for interactive testing.